src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1753,6 +1753,69 @@ internal PointerTypeSymbol CreatePointerTypeSymbol(TypeSymbol elementType)
 
         #endregion
 
+        #region Operations
+
+        /// <summary>
+        /// Gets the low-level operation corresponding to the method's body.
+        /// </summary>
+        /// <param name="method">The method symbol.</param>
+        /// <param name="cancellationToken">An optional cancellation token.</param>
+        /// <returns>The low-level operation corresponding to the method's body.</returns>
+        protected override IOperation GetOperationCore(IMethodSymbol method, CancellationToken cancellationToken = default)
+        {
+            IOperation result = null;
+            var csmethod = method.EnsureCSharpSymbolOrNull<IMethodSymbol, MethodSymbol>(nameof(method));
+
+            if (csmethod != null)
+            {
+                var body = LowerMethodBody(csmethod);
+
+                if (body != null)
+                {
+                    var operationFactory = new Semantics.CSharpOperationFactory(null);
+                    result = operationFactory.Create(body);
+                }
+            }
+
+            return result;
+        }
+
+        private BoundStatement LowerMethodBody(MethodSymbol method)
+        {
+            BoundStatement result = null;
+            var compilationState = new TypeCompilationState(method.ContainingType, this, null);
+            var diagnostics = DiagnosticBag.GetInstance();
+            var body = MethodCompiler.BindMethodBody(method, compilationState, diagnostics);
```

Comment:
>MethodCompiler.BindMethodBody [](start = 23, length = 29)

I think we should call this method only if the method symbol is SourceMemberMethodSymbol and BodySyntax is not null for it. For example, we probably don't want to return any tree for synthesized bodies (for auto-property accessors, etc.).